compiler/AST/LoopExpr.cpp

```diff
@@ -100,21 +102,22 @@ LoopExpr::LoopExpr(Expr* indices,
   gLoopExprs.add(this);
 }
 
-LoopExpr::LoopExpr(bool forall, bool zippered, bool maybeArrayType) :
+LoopExpr::LoopExpr(bool forall, bool foreach, bool zippered, bool maybeArrayType) :
```

Comment:
In here, and down there within another context there are 2-bool flag approach to determine between `for`/`foreach`/`forall`. This always feels fragile and hard to maintain for me. Can you consider:

```
enum LoopExprKind { forLoop, foreachLoop, forallLoop };
```

and replacing these bools with that enum for a more robust implementation?

Reply:
That seems to have worked :)